core/shared/src/main/scala/zio/ZManaged.scala

```diff
@@ -1481,25 +1462,25 @@ object ZManaged {
    * release action.
    */
   def finalizerExit[R](f: Exit[Any, Any] => ZIO[R, Nothing, Any]): ZManaged[R, Nothing, Unit] =
-    ZManaged.reserve(Reservation(ZIO.unit, f))
+    makeExit(ZIO.unit)((_, e) => f(e))
 
   /**
    * Creates an effect that executes a finalizer stored in a [[FinalizerRef]].
    * The `FinalizerRef` is yielded as the result of the effect, allowing for
    * control flows that require mutating finalizers.
    */
-  def finalizerRef[R](
-    initial: Exit[Any, Any] => ZIO[R, Nothing, Any]
-  ): ZManaged[R, Nothing, FinalizerRef[R]] =
-    ZManaged {
-      for {
-        finalizers <- FinalizerRef.make(initial)
-        reservation = Reservation(
-          acquire = ZIO.succeedNow(finalizers),
-          release = finalizers.runAll
-        )
-      } yield reservation
-    }
+  // def finalizerRef[R](
```

Comment:
I think it makes sense to keep this one for when you don't need the full power of ReleaseMap

Reply:
Ok, good point. The latest revision of `ZManaged.finalizerRef` actually gave you a `FinalizerRef` trait which is almost identical to `ReleaseMap`. In fact, `ReleaseMap` was based on it :-)

I think I will dial down `finalizerRef` to return something that is like a combination of `switchable` and `Ref[Finalizer]`.